### PR TITLE
Added support for getreceivedbyaddress.

### PR DESCRIPTION
**Pull request by polac**

### CODE REVIEW

File: src/fi/bittiraha/walletd/WalletRPC.java [hunk 1]
```diff
@@ -148,6 +149,18 @@ private String getnewaddress() {
     return address;
   }
 
+  private BigDecimal getreceivedbyaddress(String address) {
+    Coin retval = Coin.ZERO;
+	List<TransactionOutput> unspent = kit.wallet().calculateAllSpendCandidates(false, false);
```
Comment (jojkaart) [inline src/fi/bittiraha/walletd/WalletRPC.java]: calculateAllSpendCandidates returns a list of unspent outputs. You'll need to scan all wallet transactions to implement the equivalent of what bitcoind does.

**Review by jojkaart — CHANGES_REQUESTED** [review]

**Comment by jojkaart** [comment]
This looks almost perfect now. But it's still different from the bitcoind implementation. 
Basically two differences are still there:
1) There's no support for specifying the minconf parameter
2) bitcoind calculates the amount with minconf = 1 when it's not specified. This code does the same as minconf=0.

Could you fix these two differences? I can then merge this.

**Comment by polac** [comment]
1) added
2) It doesn't matter anymore but just wanted mention that my previous implementation was only accepting transactions which have minconf=1. I was only using pools SPENT and UNSPENT. Transaction can be in those pools only after it has at least 1 conf. Or have I misunderstood the way pools work. Anyways, doesn't matter anymore.

**Comment by jojkaart** [comment]
This pull request has now been merged. Thank you for the contribution.

**Pull request merged**